scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -114,9 +151,16 @@ private Optional<JsonObject> postRequest(final String gqlStr) throws IOException
    */
   @Override
   public long getPlanRevision(final PlanId planId) throws IOException, NoSuchPlanException, PlanServiceException {
-    final var request = "query getPlanRevision { plan_by_pk( id: %s ) { revision } }"
-        .formatted(planId.id());
-    final var response = postRequest(request).orElseThrow(() -> new NoSuchPlanException(planId));
+    final var query = """
+        query GetPlanRevision($id: Int!) {
+          plan_by_pk(id: $id) {
+            revision
+          }
+        }
+        """;
+    final var variables = Json.createObjectBuilder().add("id", planId.id()).build();
+
+    final var response = postRequest(query, variables).orElseThrow(() -> new NoSuchPlanException(planId));
```

Comment:
I believe this was already the case before this PR, so I won't insist on fixing it now, but I think that throwing NoSuchPlanException every time anything goes wrong is misleading and makes it difficult to determine what actually went wrong.

Reply:
I think in this case, it is only when the postRequest returns an empty optional, in which case it "should" interpret as the plan doesn't exist. I think the issue is in our liberal returning of Optional.empty in the postRequest methods for cases that are not actually an "empty" return value, but an error that occurred during resolution of the response (ie non 200 status codes, and interruptions)

Reply:
Generally operating fast and loose with failure modes